cmake/mcuboot.cmake

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# This file includes extra build system logic that is enabled when
+# CONFIG_MCUBOOT_SIGNATURE_KEY_FILE is set to a nonempty value.
+#
+# It builds signed binaries using imgtool as a post-processing step
+# after zephyr/zephyr.elf is created in the build directory.
+#
+# Since this file is brought in via include(), we do the work in a
+# function to avoid polluting the top-level scope.
+
+function(zephyr_mcuboot_tasks)
+  if(CONFIG_MCUBOOT_SIGNATURE_KEY_FILE STREQUAL "")
+    # No signature key file, no signed binaries. No warning is needed;
+    # this is the documented behavior.
+    return()
+  elseif(NOT EXISTS "${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}")
+    message(WARNING "Not signing images for MCUboot: CONFIG_MCUBOOT_SIGNATURE_KEY_FILE=\"${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}\" does not exist.")
+    return()
+  elseif(NOT WEST)
+    message(WARNING "Not signing images for MCUboot: west not found.")
+    return()
+  endif()
+
+  # Even though west is installed, imgtool might not be if the user has
+  # a custom manifest which doesn't include MCUboot.
+  #
+  # Therefore, go with an explicitly installed imgtool first, falling
+  # back on mcuboot/scripts/imgtool.py.
```

Comment:
Would it make sense to turn this logic upside down? If mcuboot is part of the manifest, the version of imgtool known to work with that revision of mcuboot is the one shipped with it.

Reply:
> Would it make sense to turn this logic upside down? If mcuboot is part of the manifest, the version of imgtool known to work with that revision of mcuboot is the one shipped with it.

@henrikbrixandersen My thinking here is that the user has to install imgtool with pip explicitly, because it's not part of any requirements.txt file. So the intent if it's present seems to be to use that one; doing it this way also avoids a subprocess on Windows, where they're more expensive.

If you think this is important, I will reverse the order.

Reply:
No, that's a good point. I didn't think of that. Let's keep the order as-is.